### PR TITLE
Rewrite installer & Bump to 3.0.0

### PR DESCRIPTION
**Pull request by iRebbok**
Motivation:
Installer isn't used the GitHub API, and used a standard browser link, i.e. `https://github.com/galaxy119/EXILED/releases/`, while loading the full page (no assets, but a bunch of HTML that needs parsing), this HTML parsed badly and some people were getting this instead of the download link:

```
GitHub download URL found: https://github.com/galaxy119/EXILED/releases/download/2.0.7/Exiled.Bootstrap.dll" rel="nofollow" class="d-flex flex-items-center min-width-0">
              <svg class="octicon octicon-package flex-shrink-0 text-gray" viewBox="0 0 16 16" version="1.1" width="16" height="16" aria-hidden="true"><path fill-rule="evenodd" d="M8.878.392a1.75 1.75 0 00-1.756 0l-5.25 3.045A1.75 1.75 0 001 4.951v6.098c0 .624.332 1.2.872 1.514l5.25 3.045a1.75 1.75 0 001.756 0l5.25-3.045c.54-.313.872-.89.872-1.514V4.951c0-.624-.332-1.2-.872-1.514L8.878.392zM7.875 1.69a.25.25 0 01.25 0l4.63 2.685L8 7.133 3.245 4.375l4.63-2.685zM2.5 5.677v5.372c0 .09.047.171.125.216l4.625 2.683V8.432L2.5 5.677zm6.25 8.271l4.625-2.683a.25.25 0 00.125-.216V5.677L8.75 8.432v5.516z"></path></svg>
              <span class="pl-2 flex-auto min-width-0 text-bold">Exiled.Bootstrap.dll</span>
            </a>
            <small class="pl-2 text-gray flex-shrink-0">6.5 KB</small>
          </div>
          <div class="d-flex flex-justify-between flex-items-center py-1 py-md-2 Box-body px-2">
            <a href="/galaxy119/EXILED/releases/download/2.0.7/Exiled.Installer.exe" rel="nofollow" class="d-flex flex-items-center min-width-0">
              <svg class="octicon octicon-package flex-shrink-0 text-gray" viewBox="0 0 16 16" version="1.1" width="16" height="16" aria-hidden="true"><path fill-rule="evenodd" d="M8.878.392a1.75 1.75 0 00-1.756 0l-5.25 3.045A1.75 1.75 0 001 4.951v6.098c0 .624.332 1.2.872 1.514l5.25 3.045a1.75 1.75 0 001.756 0l5.25-3.045c.54-.313.872-.89.872-1.514V4.951c0-.624-.332-1.2-.872-1.514L8.878.392zM7.875 1.69a.25.25 0 01.25 0l4.63 2.685L8 7.133 3.245 4.375l4.63-2.685zM2.5 5.677v5.372c0 .09.047.171.125.216l4.625 2.683V8.432L2.5 5.677zm6.25 8.271l4.625-2.683a.25.25 0 00.125-.216V5.677L8.75 8.432v5.516z"></path></svg>
              <span class="pl-2 flex-auto min-width-0 text-bold">Exiled.Installer.exe</span>
            </a>
            <small class="pl-2 text-gray flex-shrink-0">101 KB</small>
          </div>
          <div class="d-flex flex-justify-between flex-items-center py-1 py-md-2 Box-body px-2">
            <a href="/galaxy119/EXILED/releases/download/2.0.7/Exiled.Patcher.exe" rel="nofollow" class="d-flex flex-items-center min-width-0">
              <svg class="octicon octicon-package flex-shrink-0 text-gray" viewBox="0 0 16 16" version="1.1" width="16" height="16" aria-hidden="true"><path fill-rule="evenodd" d="M8.878.392a1.75 1.75 0 00-1.756 0l-5.25 3.045A1.75 1.75 0 001 4.951v6.098c0 .624.332 1.2.872 1.514l5.25 3.045a1.75 1.75 0 001.756 0l5.25-3.045c.54-.313.872-.89.872-1.514V4.951c0-.624-.332-1.2-.872-1.514L8.878.392zM7.875 1.69a.25.25 0 01.25 0l4.63 2.685L8 7.133 3.245 4.375l4.63-2.685zM2.5 5.677v5.372c0 .09.047.171.125.216l4.625 2.683V8.432L2.5 5.677zm6.25 8.271l4.625-2.683a.25.25 0 00.125-.216V5.677L8.75 8.432v5.516z"></path></svg>
              <span class="pl-2 flex-auto min-width-0 text-bold">Exiled.Patcher.exe</span>
            </a>
            <small class="pl-2 text-gray flex-shrink-0">512 KB</small>
          </div>
          <div class="d-flex flex-justify-between flex-items-center py-1 py-md-2 Box-body px-2">
            <a href="/galaxy119/EXILED/releases/download/2.0.7/Exiled.tar.gz, downloading...
System.Net.WebException: The remote server returned an error: (414) URI Too Long.
   at System.Net.WebClient.DownloadFile(Uri address, String fileName)
   at System.Net.WebClient.DownloadFile(String address, String fileName)
   at Exiled.Installer.Installer.Main(String[] args)
```

I rewrote the installer to .Net Core 3.1 and used [warp](https://github.com/dgiagio/warp) to package the application into a single execution file. I was unable to use CoreRT because Octokit & System.CommandLine have problems with it.

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -46,7 +47,13 @@ jobs:
     - name: Build
       env:
         EXILED_REFERENCES: ./References
-      run: msbuild EXILED.sln -p:Configuration=release
+      run: |
+        msbuild EXILED.API -p:Configuration=release
```
Comment (iopietro) [inline .github/workflows/main.yml:51]: Shouldn't `EXILED` be lowercase?

Comment (iRebbok) [reply .github/workflows/main.yml:51]: No, it finds the folder without any problems, as you can see from the successful CI.

Comment (iopietro) [reply .github/workflows/main.yml:51]: I know, because it's case insensitive, but I mean, we should name it `Exiled` to be more consistent.

Comment (iRebbok) [reply .github/workflows/main.yml:51]: Uh, I didn't notice that there is NormalCase after.

**Pull request merged**